tests/test_classification.py

```diff
@@ -634,13 +634,13 @@ def dimN_data(N):
 @pytest.mark.filterwarnings("ignore::RuntimeWarning")
 @pytest.mark.parametrize("N", [1, 3, 4])
 def test_dimN(N):
-    X, labels = dimN_data(N)
+    X, y = dimN_data(N)
     cl = CleanLearning(clf=ReshapingLogisticRegression())
     # just make sure we don't crash...
-    cl.fit(X, labels)
+    cl.fit(X, y)
     cl.predict(X)
     cl.predict_proba(X)
-    cl.score(X, labels)
+    cl.score(X, y)
 
```

Comment:
We should have one unit test that explicitly checks the named arguments `y` and `labels` with both work by trying to call cl.fit() with each of them. Let's mark this unit test with a comment:

# TODO: remove this test once `labels` is deprecated.

Reply:
You're right. Both `y` and `labels` need to be optional arguments for the deprecated signature to be valid:
```
cl.score(X=X, labels=y)
```

Two thoughts:

1. How should it be reflected in the docstring that `y` isn't "really" optional?
2. Some tests in `test_classification.py` still use `labels`, e.g. https://github.com/cleanlab/cleanlab/blob/5973a21958f47fd8e7246ccf36c8304ff4f22e3b/tests/test_classification.py#L476-L481
Should DeprecationWarnings be filtered for those tests or the signatures updated (`labels` -> `y`)? I'll add a new unit test that checks all the signature variants.